src/planner/binder/expression/bind_window_expression.cpp

```diff
@@ -142,9 +142,22 @@ static LogicalType BindRangeExpression(ClientContext &context, const string &nam
 }
 
 BindResult BaseSelectBinder::BindWindow(WindowExpression &window, idx_t depth) {
+	QueryErrorContext error_context(window.query_location);
+	//	Check for macros pretending to be aggregates
+	if (window.type == ExpressionType::WINDOW_AGGREGATE) {
+		auto func = GetCatalogEntry(CatalogType::SCALAR_FUNCTION_ENTRY, window.catalog, window.schema,
```

Comment:
Could we re-use the function we found in the catalog below, to avoid having to do two catalog look-ups in the common case?

Reply:
We seem to do multiple lookups elsewhere in the binder for similar reasons?